metrics-delivery-endpoint-impl/src/main/java/eu/openaire/mas/delivery/mapping/JsonFileBasedMappingProvider.java

```diff
@@ -142,4 +160,53 @@ private static String getInputStreamContents(InputStream is) throws IOException
             return reader.lines().collect(Collectors.joining(System.lineSeparator()));
         }
     }
+    
+    /**
+     * Watcher class responsible for handling mapping file modifications by syncing
+     * in memory mapping state with the file contents.
+     *
+     */
+    class MappingFileWatcher implements Runnable {
+        
+        private WatchService watchService;
+
+        public MappingFileWatcher(WatchService watchService) {
+            this.watchService = watchService;
+        }
+        
+        @Override
+        public void run() {
+            try {
+                WatchKey key;
+                while ((key = watchService.take()) != null) {
+                    for (WatchEvent<?> event : key.pollEvents()) {
+
+                        if (isMappingFile((Path) event.context())) {
+                            if (event.kind().equals(StandardWatchEventKinds.ENTRY_DELETE)) {
+                                log.debug("handling file related event by removing all the mappings");
```

Comment:
And this one I would probably make info.

Reply:
OK, made it an info and also changed ln 190 to keep both in line.